.github/workflows/build_workflow.yml

```diff
@@ -0,0 +1,340 @@
+name: GP2GP Build Workflow
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  checkstyle:
+    name: Checkstyle
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Create Temporary Artifacts Directory
+        run: mkdir artifacts
+
+      - name: Checkstyle
+        run: |
+          ./gradlew checkStyleMain
+          ./gradlew checkstyleTest
+        working-directory: ./service
+
+      - name: Collect Artifacts
+        run: cp -r ./service/build/reports ./artifacts
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Checkstyle Reports'
+          path: ./artifacts/**
+          compression-level: 9
```

Comment:
As per this source: [https://github.com/actions/upload-artifact](url) compression-level 9 gives the best compression but not the best speed. 
Could we maybe choose something like 5 or 6 should should give a better balance between speed and compression size? 
![image](https://github.com/user-attachments/assets/e226fe2e-c955-4559-9409-6b4f501e3781)

Reply:
Hey @ole4ryb - thanks for the comment - a useful point!
 
All artifact uploads observed in our successful runs of the workflow indicate an average of `0.6s` to run on the best compression level - given this, and due to the nature of the artifacts produced, I think I am happy to proceed with keeping it at the best compression level of `9` for now. What do you think?

Reply:
all good